sycl/source/detail/device_impl.cpp

```diff
@@ -184,7 +192,32 @@ device_impl::create_sub_devices(const std::vector<size_t> &Counts) const {
   static const cl_device_partition_property P[] = {
       CL_DEVICE_PARTITION_BY_COUNTS, CL_DEVICE_PARTITION_BY_COUNTS_LIST_END, 0};
   std::vector<cl_device_partition_property> Properties(P, P + 3);
-  Properties.insert(Properties.begin() + 1, Counts.begin(), Counts.end());
+
+  auto It = Properties.begin() + 1;
+  size_t TotalCounts = 0;
+  size_t NonZeroCounts = 0;
+  for (auto Count : Counts) {
+    TotalCounts += Count;
+    NonZeroCounts += (Count != 0) ? 1 : 0;
+    It = Properties.insert(It, Count);
+  }
```

Comment:
Please add a comment about what is happening here.

Reply:
OK